src/ScriptEditor/ui/ScriptEditorContext.tsx

```diff
@@ -28,14 +31,20 @@ export function ScriptEditorContextProvider({ children, vim }: { children: React
   const [ram, setRAM] = useState("RAM: ???");
   const [ramEntries, setRamEntries] = useState<string[][]>([["???", ""]]);
 
-  const updateRAM: ScriptEditorContextShape["updateRAM"] = (newCode, server) => {
-    if (newCode === null || server === null) {
+  const updateRAM: ScriptEditorContextShape["updateRAM"] = (newCode, filename, server) => {
+    if (newCode === null || filename === null || server === null) {
       setRAM("N/A");
       setRamEntries([["N/A", ""]]);
       return;
     }
-
-    const ramUsage = calculateRamUsage(newCode, server.scripts);
+    //temp duplication to turn FilePath into ScriptFilePath
```

Comment:
Explain this comment, please?

Reply:
the editor gets scripts and non script files (like txt)
for those something isnt included like no newCode
we cant resolve scriptFilePath before we have checked that its a script 
but after the resolving it its possible that the script is null (shouldnt happen but ts needs a check)
so i need to duplicate the same error handling like right above it because i havent found a better way yet
i planned to just make a dirty fix there and come back to it when iam (mostly/fully) done with the relative path stuff 
and then make a proper

Reply:
I think what you're looking for is to add `hasScriptExtension` to the checks above. As a type assertion, you will then be able to use the ContentFilePath as a ScrptFilePath.